src/nb_enum.cpp

```diff
@@ -150,6 +173,21 @@ bool enum_from_python(const std::type_info *tp, PyObject *o, int64_t *out, uint8
     if (!t)
         return false;
 
+    if ((t->flags & (uint32_t) enum_flags::flag_enum) != 0 && Py_TYPE(o) == t->type_py) {
+        auto pValue = PyObject_GetAttrString(o, "value");
```

Comment:
`auto` is great for complicated C++ types like (`std::map<.., ..>::const_iterator`, etc.). Here, I greatly prefer to be explicit and write `PyObject *`.